catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java

```diff
@@ -188,11 +189,11 @@ public final Attribute getAttribute(String name) {
         if (Metacard.ID.equals(name)) {
             return this.id;
         }
-        for (Attribute attribute : attributes) {
-            if (attribute == null || StringUtils.isEmpty(attribute.getName())) {
-                continue;
-            } else if (name.equals(attribute.getName())) {
-                return attribute;
+        for (Attribute attribute : Arrays.asList((attributes.toArray(new Attribute[attributes.size()])))) {
```

Comment:
It's not necessary to provide the size to the type-hinting array you pass in, so it's just as correct (and a little cheaper) to implement this as 
`for (Attribute attribute : Arrays.asList((attributes.toArray(new Attribute[0])))) {`

Otherwise, this looks great.